### PR TITLE
fix(TDP-11450): use contains filter as insensitive at runtime

### PR DESCRIPTION
**Pull request by abenali-talend**
**What is the problem this Pull Request is trying to solve?**
The contains filter is case sensitive, but we want to have a case insensitive behavior without using containsIgnoreCase. This behavior is needed for TDP when running a preparation
**What is the chosen solution to this problem?**
 Convert the contains filter from TQL to DSEL as a containIgnoreCase but only for runtime
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-11450
 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-tql/daikon-tql-dsel/src/test/java/TqlToDselConverterTest.java [hunk 1]
```diff
@@ -253,6 +253,19 @@ public void testParseContains() {
         assertEqualsForELNodes(actual, expected, true);
     }
 
+    @Test
+    public void testParseContainsForRuntime() {
```
Comment (fway87) [inline daikon-tql/daikon-tql-dsel/src/test/java/TqlToDselConverterTest.java:257]: In addition of this new test, I suggest a new test called "testParseContainsForRuntime()" in the ConvertersTogetherTest class, just after the existing test "testParseContains()"

**Comment by fway87** [comment]
The build has failed, due to a connection reset. I encounter this problem often.
Just restart the build and it will go :)

File: daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/TqlToDselVisitorForRuntime.java [hunk 2]
```diff
@@ -73,6 +73,16 @@ public ELNode visit(FieldIsInvalidExpression elt) {
         return elNode;
     }
 
+    @Override
+    public ELNode visit(FieldContainsExpression elt) {
+        ELNode containsNode = super.visit(elt);
+        // for runtime we consider the contains function as containsIgnoreCase
+        if (containsNode.getChildren().size() == 2) {
```
Comment (pgautier83) [inline daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/TqlToDselVisitorForRuntime.java:80]: why this condition?

File: daikon-tql/daikon-tql-dsel/src/test/java/TqlToDselConverterTest.java (hunk 1 above)
Comment (abenali-talend) [reply daikon-tql/daikon-tql-dsel/src/test/java/TqlToDselConverterTest.java:274]: Done

**Review by fway87 — DISMISSED** [review]
All is OK for me

**Review by pgautier83 — DISMISSED** [review]

File: daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/TqlToDselVisitorForRuntime.java [hunk 3]
```diff
@@ -119,16 +132,9 @@ private ELNode buildOrExpressionForFunctionWithAllFields(final String functionNa
...
-        TqlElement field = elt.getField();
-        ComparisonOperator operator = elt.getOperator();
-        TqlElement valueOrField = elt.getValueOrField();
-        ELNode fieldNode = field.accept(this);
-        ELNode opNode = operator.accept(this);
-        ELNode valueNode = valueOrField.accept(this);
-        opNode.addChild(fieldNode);
-        opNode.addChild(valueNode);
-        final ComparisonOperator.Enum comparisonOperator = operator.getOperator();
+        ELNode opNode = super.visit(elt);
```
Comment (ngrange-talend) [inline daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/tqltodsel/TqlToDselVisitorForRuntime.java:135]: :+1:

**Review by ngrange-talend — APPROVED** [review]

**Pull request merged**